### DISCUSSION TITLE
Automated assemble script [Batch]

### DISCUSSION THREAD

**Discussion by Pokeman2003** [vhelin/wla-dx#625] [Show and tell]
Hi again. [I made a batch file.](https://files.catbox.moe/l5wej7.bat) [Alternative Pastebin can be found here.](https://pastebin.com/9djXHymt) [Github also hosts it.](https://gist.github.com/Pokeman2003/9795ebb04e1036d0099efa765a1c898f) It can take drag and drop folders and assemble them. It's very simple and easy to use thanks to the fact that it's LITERALLY a drag and drop file, and can automate multiple folders at once.

It expects to be placed in a folder with wla-dx installed, with no alteration to the executable names. It uses the numbering part of the executable and pairs it with the file extension to determine which assembler to use(so for 68000 assembly files, use .68000, while for Z80 files, use .z80). This should make it incredibly flexible for your needs.

It also supports a level of customizability. It can execute scripts before and after assembly/linking, and also accept external attempts to "pilot" the script via label jumps. Read the comments for more explanation.

If there's any issues or glitches, please let me know so I can fix it.

**Comment by vhelin** [comment]
This might prevent the usage of this script

![image](https://github.com/vhelin/wla-dx/assets/6417746/267705a3-e680-4322-bdac-2433d03a85cc)

**Reply by Pokeman2003** [reply]
Ah, figures. Allow it through if you feel safe. I had an inkling that'd happen, so I documented the script as much as I could think of. It should provide some peace of mind being able to see the inner workings.

I have no idea why this script is quarantined but the greatly exaggerated number of bat files I've downloaded in the past few months were not.

**Reply by Pokeman2003** [reply]
Alright, threw it onto Pastebin just in case the Catbox download continues to be an issue. Hopefully though, it won't be.

**Reply by Neui** [reply]
FYI you may also want to consider [GitHub Gist](https://gist.github.com/) where you also can update the script with version tracking when you modify it.

**Reply by Pokeman2003** [reply]
Yet another tool to learn. Thanks for the heads up though. I'll probably do that for the next version.

**Comment by vhelin** [comment]
You sure the file is correct?

![image](https://github.com/vhelin/wla-dx/assets/6417746/f074334f-d9b4-4b2c-92ee-ab8f0d2691e7)

**Reply by vhelin** [reply]
Why would you name a file like that?

**Reply by Pokeman2003** [reply]
![image](https://github.com/vhelin/wla-dx/assets/83632748/e7ac0f3a-583e-4bc6-b364-2d26b831aae2)
Yeah, it's the same exact script file. Not sure what else to do. Upload it to Pastebin?

**Reply by Pokeman2003** [reply]
I didn't. Catbox just picks some random stuff to name the file so it doesn't have file name conflicts.

**Reply by vhelin** [reply]
I quickly looked through the script and didn't see it doing anything bad. Then again I'm no .bat expert. :) 

I'm not personally going to use this script as I don't really have anything work-in-progress for the 8-bit CPUs WLA DX supports, but everybody else - please share your experiences here and give feedback in this discussion!

And thanks for the script - I hope it'll be useful for WLA DX users! That Windows quarantee made me extra careful as I've been recently suggested to install CCP software on my computer without the person telling me its true origins. :) Just my paranoia then...

**Comment by vhelin** [comment]
The file in Pastebin doesn't make Windows' security to quarantee it!

**Comment by Pokeman2003** [comment]
This comment will act as the aggregate for all older versions, if you need them for whatever reason.

4/4/2024 B [Catbox](https://files.catbox.moe/l5wej7.bat) [Pastebin](https://pastebin.com/9djXHymt)
4/4/2024 OR [Catbox](https://files.catbox.moe/i6deqj.bat) [Pastebin](https://pastebin.com/bVa3kUK2)
